interfaces/builtin/fpga.go

```diff
@@ -0,0 +1,77 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const fpgaSummary = `allows access to the FPGA subsystem`
+
+const fpgaBaseDeclarationSlots = `
```

Comment:
doesn't this need a base declaration for the plug too?

i.e. I'm comparing this interface to the existing `kernel-module-control`, which is similar in privilege to this, and I see:

```go
const kernelModuleControlBaseDeclarationPlugs = `
  kernel-module-control:
    allow-installation: false
    deny-auto-connection: true
`

const kernelModuleControlBaseDeclarationSlots = `
  kernel-module-control:
    allow-installation:
      slot-snap-type:
        - core
    deny-auto-connection: true
`
```

so I think what should happen here is that:
- [ ] we should drop deny-connection from the base-decl slot
- [ ] we should add a base decl plug snippet which matches `kernel-module-control`

That being said, I do think that the base declaration system does allow the `deny-connection` to be specified just on the slot side, but I don't think that's enough of a restriction but I'm also very easily confused by the base declaration system (and thus tend to go on the side of using what's already merged as an example of what to do :wink: )

Reply:
for reference, the `kernel-module-control` interface is tested in basedelcaration_test.go under the "restricted" interfaces, and since this interface is super-privileged enough to load device trees, it should also be one of these "restricted" interfaces, but I admit to not fully understanding the distinction between the "noconnect" interfaces there and the "restricted" interfaces there

Reply:
Yes, I agree, given that this is allows to load device trees, I think it should be a superprivileged interface, that means snaps using it cannot even be installed unless there's a snap-declaration for them. To achieve this the pattern of kernel-module-control should be followed.

Reply:
@anonymouse64 I also get confused by base-declarations etc so you make a very good point to use the same pattern as kernel-module-control since this provides an equivalent level of access - thanks.

Reply:
The base declaration now looks fine and follows the correct pattern for a super-privileged interface.